main.go

```diff
@@ -46,61 +40,43 @@ type silenceResponse struct {
 }
 
 func main() {
-	clusterInit, err := getClusterCreationTime()
-	if err != nil {
-		log.Fatal(err)
-	}
-	oneHourAgo := time.Now().Add(-1 * time.Hour)
-	if clusterInit.Before(oneHourAgo) {
-		log.Printf("Cluster created more than 1 hour ago.  Exiting Cleanly.")
-		os.Exit(0)
-	}
-
-	silenceID, err := checkForExistingSilence()
-	if err != nil {
-		log.Fatal(err)
-	}
-	if silenceID == "" {
-		silenceID, err = createSilence()
+	for {
```

Comment:
In osde2e they've found that the cluster can become ready but somewhere like 5-10 minutes later it just starts upgrading, so they've created a loop where they check to see that it's healthy 20 times in a row before continuing.  Should we implement something similar here?  CC: @jeefy for additional context

Reply:
:100: -- We've often seen that a cluster will be "healthy" for a period of time then start updating operators. 

In our OSDe2e job runs we require 20 "Clean" healthchecks *_in a row_* before moving forward. 

That number was arbitrary, but I've even recently seen a cluster having 12 clean runs then going back to 0.

Reply:
> 20 "Clean" healthchecks __in a row__

over what time period?

Reply:
...looks like [30s](https://github.com/openshift/osde2e/blob/7e201f679bbe1a78b67701df77ebb69bcf768957/pkg/common/cluster/clusterutil.go#L102) x [20 by default](https://github.com/openshift/osde2e/blob/7e201f679bbe1a78b67701df77ebb69bcf768957/pkg/common/config/config.go#L540).

Reply:
I think that's fair.  At this point in the cluster lifecycle the cluster shouldn't really have much if any customer workload on it that this should disturb.

Reply:
@jeefy yesterday when I was working on the CI Watcher stuff I was looking at a cluster that had never became "healthy" after an upgrade and it was getting to 15 and even 18 before something would roll again and reset the clock.

Reply:
Yep.... :fire:

Reply:
Still working on this.

Reply:
Done via 27771f4.